Cabal/Distribution/Parsec/Types/Field.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DeriveFunctor #-}
+module Distribution.Parsec.Types.Field (
+    -- * Cabal file
+    Field (..),
+    fieldAnn,
+    FieldLine (..),
+    SectionArg (..),
+    sectionArgAnn,
+    -- * Name
+    Name (..),
+    mkName,
+    getName,
+    nameAnn,
+    ) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Data.ByteString (ByteString)
+
+import qualified Data.Char as Char
+import qualified Data.ByteString.Char8 as B
+
+-------------------------------------------------------------------------------
+-- Cabal file
+-------------------------------------------------------------------------------
+
+data Field ann
```

Comment:
-- | A Cabal file consists of a series of fields (@foo: bar@) and sections (@library ...@).